exercises/main.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+func insertion(s []int) {
+	for i := 1; i < len(s); i++ {
+		for j := i; j > 0; j-- {
+			if s[j-1] > s[j] {
+				s[j-1], s[j] = s[j], s[j-1]
+			}
+		}
+	}
+}
+func selection(s []int) {
+	for i := 0; i < len(s); i++ {
+		for j, min := i+1, i; j < len(s); j++ {
+			if s[j] < s[min] {
+				s[j], s[min] = s[min], s[j]
```

Comment:
this algorithm should not swap multiple times. it should find the smallest (or the largest, depending on your implementation) and produce only one swap. pseudo-code would look like this:
```
for smallest in indices of s {
  index of min = find index of min in s[smallest:]
  if index of min is not smallest {
    swap index of min and smallest in s
  }
}
```
it's important there is only one swap